docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,429 @@
+staged or unstaged, run:
+
+```
+git status
+```
+
+To stage all untracked files, run:
+
+```
+git add --all
+```
+
+To stage one untracked file at a time, run:
+
+```
+git add  { relative path to file from root directory }
+```
+
+When you are pleased with your staged changes, it is time to create a `commit`
+and give it a message.
+
+Please read the
+[Commit Message Format](https://loopback.io/doc/en/contrib/code-contrib-lb4.html#commit-message-guidelines)
+guidelines to correctly format your commit message.
+
+To help with abiding by the rules of commit messages, please use the
+`commitizen` tool mentioned in the documentation above. This means will we use
+`git cz` instead of `git commit`. Install `commitizen` by running
+`npm install -g commitizen`.
+
+In my case, I want the commit message to look like this:
+
+```
+docs: tutorial on how to submit a pull request to lb4
+
+This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+
+```
+
+To start `commitizen`, run :
+
+```
+git cz
+```
+
+Follow the prompts for the various fields.
+
+In my case I have entered :
+
+```
+type                : docs
+scope               : <blank>
+short description   : tutorial on how to submit a pull request to lb4
+long description    : This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+breaking changes    : N
+affects open issues : N
+```
+
+![submit_pr_git_cz_1.png](./imgs/submit_pr_git_cz_1.png)
+
+The interactive `commitizen` prompts complete and the commit is created with a
+properly formatted message.
+
+### 5. Push commits to the remote feature branch
+
+It is now time to push your local committed changes of your local feature branch
+directory to your remote feature branch; to keep them in synch.
+
+The very `first` time you push your changes you should run:
+
+```
+git push --set-upstream origin emonddr-doc-changes
+```
+
+The `--set-upstream` parameter will set up the tracking between your local
+feature branch and your remote feature branch.
+
+Afterwards, for any additional push, you just need to run:
+
+```
+git push
+```
+
+It is not necessary to create a pull request immediately on the push of your
+first commit; this can be done later.
+
+Do take some time to think about how many commits you want in the pull request
+when you eventually create it.
+
+If it makes sense to place your changes into one commit, then do so.
+
+If, however, it makes more sense to place your changes into a few logically
+separate commits, then do so.
+
+Whatever is the best way for the maintainers to understand the changes when the
+pull request is being reviewed.
+
+You can always re-organize commits or compress the number of commits later
+through a step called `Squashing Commits`.
+
+### 6. Rebasing
+
+Eventually your fork of the original repository will become stale, and it will
```

Comment:
maybe use `if` instead of `eventually`?  because someone will be "lucky" enough that the PR got merged before it becomes stale. :)

Reply:
Suggested changes: 
If your fork becomes stale at any point in time, you can bring it up-to-date by rebasing your PR. 
```
<here are the steps>
```
then it goes directly to ur next snippet `git remote...`

Reply:
+1. `stale` is a bit confusing too. Maybe we should say `your branch is behind the original/master`.